server/channels/app/plugin_install_test.go

```diff
@@ -265,6 +265,109 @@ func TestInstallPluginLocally(t *testing.T) {
 			assertBundleInfoManifests(t, th, []*model.Manifest{manifest})
 		})
 	})
+
+	installPluginUpdateConfig := func(t *testing.T, th *TestHelper, id, version string, installationStrategy pluginInstallationStrategy) (*model.Manifest, map[string]any, *model.AppError) {
+		t.Helper()
+		var mp map[string]any
+
+		manifest := &model.Manifest{
+			Id:      id,
+			Version: version,
+		}
+		manifestJSON, jsonErr := json.Marshal(manifest)
+		require.NoError(t, jsonErr)
+		reader := makeInMemoryGzipTarFile(t, []testFile{
+			{"plugin.json", string(manifestJSON)},
+		})
+
+		actualManifest, appError := th.App.ch.installPluginLocally(reader, installationStrategy)
+		require.Nil(t, appError)
+		require.Equal(t, manifest, actualManifest)
+
+		mp = th.App.Config().PluginSettings.Plugins[actualManifest.Id]
+		return actualManifest, mp, appError
+	}
+
+	t.Run("Config updates because manifest ID does not exist in map", func(t *testing.T) {
+		th := Setup(t)
+		defer th.TearDown()
+		cleanExistingBundles(t, th)
+		expectedMap := make(map[string]any)
+		expectedMap["id"] = "valid"
+		expectedMap["version"] = "0.0.2"
+
+		_, returnedMap, appErr := installPluginUpdateConfig(t, th, "valid", "0.0.2", installPluginLocallyAlways)
+		require.Nil(t, appErr)
+
+		//will probably have to make a separte pluginSetting struct with dummy vals to check against return plugin settign vals
+		for k, v := range returnedMap {
+			require.Equal(t, expectedMap[k], v)
+		}
+	})
+
+	t.Run("Config does not update because manifest ID already exist in map", func(t *testing.T) {
+		th := Setup(t)
+		defer th.TearDown()
+		cleanExistingBundles(t, th)
+
+		//create Plugin: `myplugin`
+		setupPluginAPITest(t, ` 
+		package main 
+ 
+		import ( 
+			"net/http" 
+			"encoding/json" 
+ 
+			"github.com/mattermost/mattermost/server/public/plugin" 
+			"github.com/mattermost/mattermost/server/public/model" 
+		) 
+ 
+		type MyPlugin struct {
+			plugin.MattermostPlugin,
+			enableUpload bool,
+			hasRootAccess bool,
+			likesPie bool,
+			version string,
+			id string
```

Comment:
Go struct fields are not separated by commata
```suggestion
			plugin.MattermostPlugin
			enableUpload  bool
			hasRootAccess bool
			likesPie      bool
			version       string
			id            string
```

Reply:
damn 😞 

My bad.  When I ran the e2e tests locally it said nothing about this.